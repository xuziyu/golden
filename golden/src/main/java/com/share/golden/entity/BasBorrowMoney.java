package com.share.golden.entity;

public class BasBorrowMoney {
    private String borrowId;

    private String borrowSerial;

    private String userId;

    private String borrowCompany;

    private String borrowMoney;

    private String borrowRate;

    private String timeLimit;

    private String borrowLimit;

    private String borrowState;

    private String borrowRecommend;

    private String borrowType;

    private String borrowRemark1;

    private String borrowRemark2;

    public String getBorrowId() {
        return borrowId;
    }

    public void setBorrowId(String borrowId) {
        this.borrowId = borrowId == null ? null : borrowId.trim();
    }

    public String getBorrowSerial() {
        return borrowSerial;
    }

    public void setBorrowSerial(String borrowSerial) {
        this.borrowSerial = borrowSerial == null ? null : borrowSerial.trim();
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId == null ? null : userId.trim();
    }

    public String getBorrowCompany() {
        return borrowCompany;
    }

    public void setBorrowCompany(String borrowCompany) {
        this.borrowCompany = borrowCompany == null ? null : borrowCompany.trim();
    }

    public String getBorrowMoney() {
        return borrowMoney;
    }

    public void setBorrowMoney(String borrowMoney) {
        this.borrowMoney = borrowMoney == null ? null : borrowMoney.trim();
    }

    public String getBorrowRate() {
        return borrowRate;
    }

    public void setBorrowRate(String borrowRate) {
        this.borrowRate = borrowRate == null ? null : borrowRate.trim();
    }

    public String getTimeLimit() {
        return timeLimit;
    }

    public void setTimeLimit(String timeLimit) {
        this.timeLimit = timeLimit == null ? null : timeLimit.trim();
    }

    public String getBorrowLimit() {
        return borrowLimit;
    }

    public void setBorrowLimit(String borrowLimit) {
        this.borrowLimit = borrowLimit == null ? null : borrowLimit.trim();
    }

    public String getBorrowState() {
        return borrowState;
    }

    public void setBorrowState(String borrowState) {
        this.borrowState = borrowState == null ? null : borrowState.trim();
    }

    public String getBorrowRecommend() {
        return borrowRecommend;
    }

    public void setBorrowRecommend(String borrowRecommend) {
        this.borrowRecommend = borrowRecommend == null ? null : borrowRecommend.trim();
    }

    public String getBorrowType() {
        return borrowType;
    }

    public void setBorrowType(String borrowType) {
        this.borrowType = borrowType == null ? null : borrowType.trim();
    }

    public String getBorrowRemark1() {
        return borrowRemark1;
    }

    public void setBorrowRemark1(String borrowRemark1) {
        this.borrowRemark1 = borrowRemark1 == null ? null : borrowRemark1.trim();
    }

    public String getBorrowRemark2() {
        return borrowRemark2;
    }

    public void setBorrowRemark2(String borrowRemark2) {
        this.borrowRemark2 = borrowRemark2 == null ? null : borrowRemark2.trim();
    }
}