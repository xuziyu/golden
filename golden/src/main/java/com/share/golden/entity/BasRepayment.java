package com.share.golden.entity;

import java.util.Date;

public class BasRepayment {
    private String repayId;

    private String repayCode;

    private Date repayDate;

    private String repayState;

    private String repayTime;

    private String repayRemark1;

    private String repayRemark2;

    public String getRepayId() {
        return repayId;
    }

    public void setRepayId(String repayId) {
        this.repayId = repayId == null ? null : repayId.trim();
    }

    public String getRepayCode() {
        return repayCode;
    }

    public void setRepayCode(String repayCode) {
        this.repayCode = repayCode == null ? null : repayCode.trim();
    }

    public Date getRepayDate() {
        return repayDate;
    }

    public void setRepayDate(Date repayDate) {
        this.repayDate = repayDate;
    }

    public String getRepayState() {
        return repayState;
    }

    public void setRepayState(String repayState) {
        this.repayState = repayState == null ? null : repayState.trim();
    }

    public String getRepayTime() {
        return repayTime;
    }

    public void setRepayTime(String repayTime) {
        this.repayTime = repayTime == null ? null : repayTime.trim();
    }

    public String getRepayRemark1() {
        return repayRemark1;
    }

    public void setRepayRemark1(String repayRemark1) {
        this.repayRemark1 = repayRemark1 == null ? null : repayRemark1.trim();
    }

    public String getRepayRemark2() {
        return repayRemark2;
    }

    public void setRepayRemark2(String repayRemark2) {
        this.repayRemark2 = repayRemark2 == null ? null : repayRemark2.trim();
    }
}