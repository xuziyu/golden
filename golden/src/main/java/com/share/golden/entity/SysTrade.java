package com.share.golden.entity;

import java.util.Date;

public class SysTrade {
	private String tradeId;

	private String userI;

	private String tradeMoney;

	private String tradeType;

	private Date tradeTime;

	private String tradeRemark;

	public String getTradeId() {
		return tradeId;
	}

	public void setTradeId(String tradeId) {
		this.tradeId = tradeId == null ? null : tradeId.trim();
	}

	public String getUserI() {
		return userI;
	}

	public void setUserI(String userI) {
		this.userI = userI == null ? null : userI.trim();
	}

	public String getTradeMoney() {
		return tradeMoney;
	}

	public void setTradeMoney(String tradeMoney) {
		this.tradeMoney = tradeMoney == null ? null : tradeMoney.trim();
	}

	public String getTradeType() {
		return tradeType;
	}

	public void setTradeType(String tradeType) {
		this.tradeType = tradeType == null ? null : tradeType.trim();
	}

	public Date getTradeTime() {
		return tradeTime;
	}

	public void setTradeTime(Date tradeTime) {
		this.tradeTime = tradeTime;
	}

	public String getTradeRemark() {
		return tradeRemark;
	}

	public void setTradeRemark(String tradeRemark) {
		this.tradeRemark = tradeRemark == null ? null : tradeRemark.trim();
	}
}