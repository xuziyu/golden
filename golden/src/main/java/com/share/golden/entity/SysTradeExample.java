package com.share.golden.entity;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class SysTradeExample {
    protected String orderByClause;

    protected boolean distinct;

    protected List<Criteria> oredCriteria;

    public SysTradeExample() {
        oredCriteria = new ArrayList<Criteria>();
    }

    public void setOrderByClause(String orderByClause) {
        this.orderByClause = orderByClause;
    }

    public String getOrderByClause() {
        return orderByClause;
    }

    public void setDistinct(boolean distinct) {
        this.distinct = distinct;
    }

    public boolean isDistinct() {
        return distinct;
    }

    public List<Criteria> getOredCriteria() {
        return oredCriteria;
    }

    public void or(Criteria criteria) {
        oredCriteria.add(criteria);
    }

    public Criteria or() {
        Criteria criteria = createCriteriaInternal();
        oredCriteria.add(criteria);
        return criteria;
    }

    public Criteria createCriteria() {
        Criteria criteria = createCriteriaInternal();
        if (oredCriteria.size() == 0) {
            oredCriteria.add(criteria);
        }
        return criteria;
    }

    protected Criteria createCriteriaInternal() {
        Criteria criteria = new Criteria();
        return criteria;
    }

    public void clear() {
        oredCriteria.clear();
        orderByClause = null;
        distinct = false;
    }

    protected abstract static class GeneratedCriteria {
        protected List<Criterion> criteria;

        protected GeneratedCriteria() {
            super();
            criteria = new ArrayList<Criterion>();
        }

        public boolean isValid() {
            return criteria.size() > 0;
        }

        public List<Criterion> getAllCriteria() {
            return criteria;
        }

        public List<Criterion> getCriteria() {
            return criteria;
        }

        protected void addCriterion(String condition) {
            if (condition == null) {
                throw new RuntimeException("Value for condition cannot be null");
            }
            criteria.add(new Criterion(condition));
        }

        protected void addCriterion(String condition, Object value, String property) {
            if (value == null) {
                throw new RuntimeException("Value for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value));
        }

        protected void addCriterion(String condition, Object value1, Object value2, String property) {
            if (value1 == null || value2 == null) {
                throw new RuntimeException("Between values for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value1, value2));
        }

        public Criteria andTradeIdIsNull() {
            addCriterion("\"TRADE_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andTradeIdIsNotNull() {
            addCriterion("\"TRADE_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andTradeIdEqualTo(String value) {
            addCriterion("\"TRADE_ID\" =", value, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdNotEqualTo(String value) {
            addCriterion("\"TRADE_ID\" <>", value, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdGreaterThan(String value) {
            addCriterion("\"TRADE_ID\" >", value, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdGreaterThanOrEqualTo(String value) {
            addCriterion("\"TRADE_ID\" >=", value, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdLessThan(String value) {
            addCriterion("\"TRADE_ID\" <", value, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdLessThanOrEqualTo(String value) {
            addCriterion("\"TRADE_ID\" <=", value, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdLike(String value) {
            addCriterion("\"TRADE_ID\" like", value, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdNotLike(String value) {
            addCriterion("\"TRADE_ID\" not like", value, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdIn(List<String> values) {
            addCriterion("\"TRADE_ID\" in", values, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdNotIn(List<String> values) {
            addCriterion("\"TRADE_ID\" not in", values, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdBetween(String value1, String value2) {
            addCriterion("\"TRADE_ID\" between", value1, value2, "tradeId");
            return (Criteria) this;
        }

        public Criteria andTradeIdNotBetween(String value1, String value2) {
            addCriterion("\"TRADE_ID\" not between", value1, value2, "tradeId");
            return (Criteria) this;
        }

        public Criteria andUserIIsNull() {
            addCriterion("\"USER_I\" is null");
            return (Criteria) this;
        }

        public Criteria andUserIIsNotNull() {
            addCriterion("\"USER_I\" is not null");
            return (Criteria) this;
        }

        public Criteria andUserIEqualTo(String value) {
            addCriterion("\"USER_I\" =", value, "userI");
            return (Criteria) this;
        }

        public Criteria andUserINotEqualTo(String value) {
            addCriterion("\"USER_I\" <>", value, "userI");
            return (Criteria) this;
        }

        public Criteria andUserIGreaterThan(String value) {
            addCriterion("\"USER_I\" >", value, "userI");
            return (Criteria) this;
        }

        public Criteria andUserIGreaterThanOrEqualTo(String value) {
            addCriterion("\"USER_I\" >=", value, "userI");
            return (Criteria) this;
        }

        public Criteria andUserILessThan(String value) {
            addCriterion("\"USER_I\" <", value, "userI");
            return (Criteria) this;
        }

        public Criteria andUserILessThanOrEqualTo(String value) {
            addCriterion("\"USER_I\" <=", value, "userI");
            return (Criteria) this;
        }

        public Criteria andUserILike(String value) {
            addCriterion("\"USER_I\" like", value, "userI");
            return (Criteria) this;
        }

        public Criteria andUserINotLike(String value) {
            addCriterion("\"USER_I\" not like", value, "userI");
            return (Criteria) this;
        }

        public Criteria andUserIIn(List<String> values) {
            addCriterion("\"USER_I\" in", values, "userI");
            return (Criteria) this;
        }

        public Criteria andUserINotIn(List<String> values) {
            addCriterion("\"USER_I\" not in", values, "userI");
            return (Criteria) this;
        }

        public Criteria andUserIBetween(String value1, String value2) {
            addCriterion("\"USER_I\" between", value1, value2, "userI");
            return (Criteria) this;
        }

        public Criteria andUserINotBetween(String value1, String value2) {
            addCriterion("\"USER_I\" not between", value1, value2, "userI");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyIsNull() {
            addCriterion("\"TRADE_MONEY\" is null");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyIsNotNull() {
            addCriterion("\"TRADE_MONEY\" is not null");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyEqualTo(String value) {
            addCriterion("\"TRADE_MONEY\" =", value, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyNotEqualTo(String value) {
            addCriterion("\"TRADE_MONEY\" <>", value, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyGreaterThan(String value) {
            addCriterion("\"TRADE_MONEY\" >", value, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyGreaterThanOrEqualTo(String value) {
            addCriterion("\"TRADE_MONEY\" >=", value, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyLessThan(String value) {
            addCriterion("\"TRADE_MONEY\" <", value, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyLessThanOrEqualTo(String value) {
            addCriterion("\"TRADE_MONEY\" <=", value, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyLike(String value) {
            addCriterion("\"TRADE_MONEY\" like", value, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyNotLike(String value) {
            addCriterion("\"TRADE_MONEY\" not like", value, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyIn(List<String> values) {
            addCriterion("\"TRADE_MONEY\" in", values, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyNotIn(List<String> values) {
            addCriterion("\"TRADE_MONEY\" not in", values, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyBetween(String value1, String value2) {
            addCriterion("\"TRADE_MONEY\" between", value1, value2, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeMoneyNotBetween(String value1, String value2) {
            addCriterion("\"TRADE_MONEY\" not between", value1, value2, "tradeMoney");
            return (Criteria) this;
        }

        public Criteria andTradeTypeIsNull() {
            addCriterion("\"TRADE_TYPE\" is null");
            return (Criteria) this;
        }

        public Criteria andTradeTypeIsNotNull() {
            addCriterion("\"TRADE_TYPE\" is not null");
            return (Criteria) this;
        }

        public Criteria andTradeTypeEqualTo(String value) {
            addCriterion("\"TRADE_TYPE\" =", value, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeNotEqualTo(String value) {
            addCriterion("\"TRADE_TYPE\" <>", value, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeGreaterThan(String value) {
            addCriterion("\"TRADE_TYPE\" >", value, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeGreaterThanOrEqualTo(String value) {
            addCriterion("\"TRADE_TYPE\" >=", value, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeLessThan(String value) {
            addCriterion("\"TRADE_TYPE\" <", value, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeLessThanOrEqualTo(String value) {
            addCriterion("\"TRADE_TYPE\" <=", value, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeLike(String value) {
            addCriterion("\"TRADE_TYPE\" like", value, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeNotLike(String value) {
            addCriterion("\"TRADE_TYPE\" not like", value, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeIn(List<String> values) {
            addCriterion("\"TRADE_TYPE\" in", values, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeNotIn(List<String> values) {
            addCriterion("\"TRADE_TYPE\" not in", values, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeBetween(String value1, String value2) {
            addCriterion("\"TRADE_TYPE\" between", value1, value2, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTypeNotBetween(String value1, String value2) {
            addCriterion("\"TRADE_TYPE\" not between", value1, value2, "tradeType");
            return (Criteria) this;
        }

        public Criteria andTradeTimeIsNull() {
            addCriterion("\"TRADE_TIME\" is null");
            return (Criteria) this;
        }

        public Criteria andTradeTimeIsNotNull() {
            addCriterion("\"TRADE_TIME\" is not null");
            return (Criteria) this;
        }

        public Criteria andTradeTimeEqualTo(Date value) {
            addCriterion("\"TRADE_TIME\" =", value, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeTimeNotEqualTo(Date value) {
            addCriterion("\"TRADE_TIME\" <>", value, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeTimeGreaterThan(Date value) {
            addCriterion("\"TRADE_TIME\" >", value, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeTimeGreaterThanOrEqualTo(Date value) {
            addCriterion("\"TRADE_TIME\" >=", value, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeTimeLessThan(Date value) {
            addCriterion("\"TRADE_TIME\" <", value, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeTimeLessThanOrEqualTo(Date value) {
            addCriterion("\"TRADE_TIME\" <=", value, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeTimeIn(List<Date> values) {
            addCriterion("\"TRADE_TIME\" in", values, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeTimeNotIn(List<Date> values) {
            addCriterion("\"TRADE_TIME\" not in", values, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeTimeBetween(Date value1, Date value2) {
            addCriterion("\"TRADE_TIME\" between", value1, value2, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeTimeNotBetween(Date value1, Date value2) {
            addCriterion("\"TRADE_TIME\" not between", value1, value2, "tradeTime");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkIsNull() {
            addCriterion("\"TRADE_REMARK\" is null");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkIsNotNull() {
            addCriterion("\"TRADE_REMARK\" is not null");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkEqualTo(String value) {
            addCriterion("\"TRADE_REMARK\" =", value, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkNotEqualTo(String value) {
            addCriterion("\"TRADE_REMARK\" <>", value, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkGreaterThan(String value) {
            addCriterion("\"TRADE_REMARK\" >", value, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkGreaterThanOrEqualTo(String value) {
            addCriterion("\"TRADE_REMARK\" >=", value, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkLessThan(String value) {
            addCriterion("\"TRADE_REMARK\" <", value, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkLessThanOrEqualTo(String value) {
            addCriterion("\"TRADE_REMARK\" <=", value, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkLike(String value) {
            addCriterion("\"TRADE_REMARK\" like", value, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkNotLike(String value) {
            addCriterion("\"TRADE_REMARK\" not like", value, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkIn(List<String> values) {
            addCriterion("\"TRADE_REMARK\" in", values, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkNotIn(List<String> values) {
            addCriterion("\"TRADE_REMARK\" not in", values, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkBetween(String value1, String value2) {
            addCriterion("\"TRADE_REMARK\" between", value1, value2, "tradeRemark");
            return (Criteria) this;
        }

        public Criteria andTradeRemarkNotBetween(String value1, String value2) {
            addCriterion("\"TRADE_REMARK\" not between", value1, value2, "tradeRemark");
            return (Criteria) this;
        }
    }

    public static class Criteria extends GeneratedCriteria {

        protected Criteria() {
            super();
        }
    }

    public static class Criterion {
        private String condition;

        private Object value;

        private Object secondValue;

        private boolean noValue;

        private boolean singleValue;

        private boolean betweenValue;

        private boolean listValue;

        private String typeHandler;

        public String getCondition() {
            return condition;
        }

        public Object getValue() {
            return value;
        }

        public Object getSecondValue() {
            return secondValue;
        }

        public boolean isNoValue() {
            return noValue;
        }

        public boolean isSingleValue() {
            return singleValue;
        }

        public boolean isBetweenValue() {
            return betweenValue;
        }

        public boolean isListValue() {
            return listValue;
        }

        public String getTypeHandler() {
            return typeHandler;
        }

        protected Criterion(String condition) {
            super();
            this.condition = condition;
            this.typeHandler = null;
            this.noValue = true;
        }

        protected Criterion(String condition, Object value, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.typeHandler = typeHandler;
            if (value instanceof List<?>) {
                this.listValue = true;
            } else {
                this.singleValue = true;
            }
        }

        protected Criterion(String condition, Object value) {
            this(condition, value, null);
        }

        protected Criterion(String condition, Object value, Object secondValue, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.secondValue = secondValue;
            this.typeHandler = typeHandler;
            this.betweenValue = true;
        }

        protected Criterion(String condition, Object value, Object secondValue) {
            this(condition, value, secondValue, null);
        }
    }
}