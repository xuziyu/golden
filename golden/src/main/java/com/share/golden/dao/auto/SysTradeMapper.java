package com.share.golden.dao.auto;

import com.share.golden.entity.SysTrade;
import com.share.golden.entity.SysTradeExample;
import java.util.List;
import org.apache.ibatis.annotations.Param;

public interface SysTradeMapper {
    long countByExample(SysTradeExample example);

    int deleteByExample(SysTradeExample example);

    int deleteByPrimaryKey(String tradeId);

    int insert(SysTrade record);

    int insertSelective(SysTrade record);

    List<SysTrade> selectByExample(SysTradeExample example);

    SysTrade selectByPrimaryKey(String tradeId);

    int updateByExampleSelective(@Param("record") SysTrade record, @Param("example") SysTradeExample example);

    int updateByExample(@Param("record") SysTrade record, @Param("example") SysTradeExample example);

    int updateByPrimaryKeySelective(SysTrade record);

    int updateByPrimaryKey(SysTrade record);
}