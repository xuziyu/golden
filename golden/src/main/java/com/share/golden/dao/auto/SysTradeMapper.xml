<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
<mapper namespace="com.share.golden.dao.auto.SysTradeMapper">
  <resultMap id="BaseResultMap" type="com.share.golden.entity.SysTrade">
    <id column="TRADE_ID" jdbcType="VARCHAR" property="tradeId" />
    <result column="USER_I" jdbcType="VARCHAR" property="userI" />
    <result column="TRADE_MONEY" jdbcType="VARCHAR" property="tradeMoney" />
    <result column="TRADE_TYPE" jdbcType="VARCHAR" property="tradeType" />
    <result column="TRADE_TIME" jdbcType="TIMESTAMP" property="tradeTime" />
    <result column="TRADE_REMARK" jdbcType="VARCHAR" property="tradeRemark" />
  </resultMap>
  <sql id="Example_Where_Clause">
    <where>
      <foreach collection="oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Update_By_Example_Where_Clause">
    <where>
      <foreach collection="example.oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Base_Column_List">
    "TRADE_ID", "USER_I", "TRADE_MONEY", "TRADE_TYPE", "TRADE_TIME", "TRADE_REMARK"
  </sql>
  <select id="selectByExample" parameterType="com.share.golden.entity.SysTradeExample" resultMap="BaseResultMap">
    select
    <if test="distinct">
      distinct
    </if>
    <include refid="Base_Column_List" />
    from SYS_TRADE
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
    <if test="orderByClause != null">
      order by ${orderByClause}
    </if>
  </select>
  <select id="selectByPrimaryKey" parameterType="java.lang.String" resultMap="BaseResultMap">
    select 
    <include refid="Base_Column_List" />
    from SYS_TRADE
    where "TRADE_ID" = #{tradeId,jdbcType=VARCHAR}
  </select>
  <delete id="deleteByPrimaryKey" parameterType="java.lang.String">
    delete from SYS_TRADE
    where "TRADE_ID" = #{tradeId,jdbcType=VARCHAR}
  </delete>
  <delete id="deleteByExample" parameterType="com.share.golden.entity.SysTradeExample">
    delete from SYS_TRADE
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </delete>
  <insert id="insert" parameterType="com.share.golden.entity.SysTrade">
    insert into SYS_TRADE ("TRADE_ID", "USER_I", "TRADE_MONEY", 
      "TRADE_TYPE", "TRADE_TIME", "TRADE_REMARK"
      )
    values (#{tradeId,jdbcType=VARCHAR}, #{userI,jdbcType=VARCHAR}, #{tradeMoney,jdbcType=VARCHAR}, 
      #{tradeType,jdbcType=VARCHAR}, #{tradeTime,jdbcType=TIMESTAMP}, #{tradeRemark,jdbcType=VARCHAR}
      )
  </insert>
  <insert id="insertSelective" parameterType="com.share.golden.entity.SysTrade">
    insert into SYS_TRADE
    <trim prefix="(" suffix=")" suffixOverrides=",">
      <if test="tradeId != null">
        "TRADE_ID",
      </if>
      <if test="userI != null">
        "USER_I",
      </if>
      <if test="tradeMoney != null">
        "TRADE_MONEY",
      </if>
      <if test="tradeType != null">
        "TRADE_TYPE",
      </if>
      <if test="tradeTime != null">
        "TRADE_TIME",
      </if>
      <if test="tradeRemark != null">
        "TRADE_REMARK",
      </if>
    </trim>
    <trim prefix="values (" suffix=")" suffixOverrides=",">
      <if test="tradeId != null">
        #{tradeId,jdbcType=VARCHAR},
      </if>
      <if test="userI != null">
        #{userI,jdbcType=VARCHAR},
      </if>
      <if test="tradeMoney != null">
        #{tradeMoney,jdbcType=VARCHAR},
      </if>
      <if test="tradeType != null">
        #{tradeType,jdbcType=VARCHAR},
      </if>
      <if test="tradeTime != null">
        #{tradeTime,jdbcType=TIMESTAMP},
      </if>
      <if test="tradeRemark != null">
        #{tradeRemark,jdbcType=VARCHAR},
      </if>
    </trim>
  </insert>
  <select id="countByExample" parameterType="com.share.golden.entity.SysTradeExample" resultType="java.lang.Long">
    select count(*) from SYS_TRADE
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </select>
  <update id="updateByExampleSelective" parameterType="map">
    update SYS_TRADE
    <set>
      <if test="record.tradeId != null">
        "TRADE_ID" = #{record.tradeId,jdbcType=VARCHAR},
      </if>
      <if test="record.userI != null">
        "USER_I" = #{record.userI,jdbcType=VARCHAR},
      </if>
      <if test="record.tradeMoney != null">
        "TRADE_MONEY" = #{record.tradeMoney,jdbcType=VARCHAR},
      </if>
      <if test="record.tradeType != null">
        "TRADE_TYPE" = #{record.tradeType,jdbcType=VARCHAR},
      </if>
      <if test="record.tradeTime != null">
        "TRADE_TIME" = #{record.tradeTime,jdbcType=TIMESTAMP},
      </if>
      <if test="record.tradeRemark != null">
        "TRADE_REMARK" = #{record.tradeRemark,jdbcType=VARCHAR},
      </if>
    </set>
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByExample" parameterType="map">
    update SYS_TRADE
    set "TRADE_ID" = #{record.tradeId,jdbcType=VARCHAR},
      "USER_I" = #{record.userI,jdbcType=VARCHAR},
      "TRADE_MONEY" = #{record.tradeMoney,jdbcType=VARCHAR},
      "TRADE_TYPE" = #{record.tradeType,jdbcType=VARCHAR},
      "TRADE_TIME" = #{record.tradeTime,jdbcType=TIMESTAMP},
      "TRADE_REMARK" = #{record.tradeRemark,jdbcType=VARCHAR}
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByPrimaryKeySelective" parameterType="com.share.golden.entity.SysTrade">
    update SYS_TRADE
    <set>
      <if test="userI != null">
        "USER_I" = #{userI,jdbcType=VARCHAR},
      </if>
      <if test="tradeMoney != null">
        "TRADE_MONEY" = #{tradeMoney,jdbcType=VARCHAR},
      </if>
      <if test="tradeType != null">
        "TRADE_TYPE" = #{tradeType,jdbcType=VARCHAR},
      </if>
      <if test="tradeTime != null">
        "TRADE_TIME" = #{tradeTime,jdbcType=TIMESTAMP},
      </if>
      <if test="tradeRemark != null">
        "TRADE_REMARK" = #{tradeRemark,jdbcType=VARCHAR},
      </if>
    </set>
    where "TRADE_ID" = #{tradeId,jdbcType=VARCHAR}
  </update>
  <update id="updateByPrimaryKey" parameterType="com.share.golden.entity.SysTrade">
    update SYS_TRADE
    set "USER_I" = #{userI,jdbcType=VARCHAR},
      "TRADE_MONEY" = #{tradeMoney,jdbcType=VARCHAR},
      "TRADE_TYPE" = #{tradeType,jdbcType=VARCHAR},
      "TRADE_TIME" = #{tradeTime,jdbcType=TIMESTAMP},
      "TRADE_REMARK" = #{tradeRemark,jdbcType=VARCHAR}
    where "TRADE_ID" = #{tradeId,jdbcType=VARCHAR}
  </update>
</mapper>